src/components/SignUp.js

```diff
@@ -1,11 +1,122 @@
-const SignUp = () => {
+import { Link, useHistory } from 'react-router-dom'
+import { useState } from 'react'
+import { FaCameraRetro } from 'react-icons/fa'
+import signupImg1 from '../assets/signupImg1.jpg'
+
+const SignUp = ({ onSignup }) => {
+    let history = useHistory()
+    const [username, setUsername] = useState('')
+    const [password, setPassword] = useState('')
+    const [first_name, setFirstName] = useState('')
+    const [last_name, setLastName] = useState('')
+    const [email, setEmail] = useState('')
+    const [admin, setAdmin] = useState(false)
+    const [errorUserName, setErrorUsername] = useState(false)
+    const [errorInputBlank, setErrorInput] = useState(false)
+
+    const onSubmit = (e) => {
+        e.preventDefault()
+        // removes any previous error from screen when successful login to dashboard
+        setErrorInput(false)
+        setErrorUsername(false)
+
+        // username taken 
+        if (username == username) {
+            console.log("Username isse")
+            // removes any previous error from screen
+            setErrorInput(false)
+            setErrorUsername(true)
+        }
+
+        // blank input
+        if (username == '' || password == '' || first_name == '' || last_name == '' || email == '') {
+            // removes any previous error from screen
+            setErrorUsername(false)
+            setErrorInput(true)
+        }
```

Comment:
If there are blank inputs you should return here so that you won't continue and hit the endpoint and the user can add input so they can Signup correctly

Reply:
That makes sense, that you for letting me know. I'll add the return